src/postformatting/rule-selector-remover.js

```diff
@@ -0,0 +1,32 @@
+import csstree from 'css-tree'
+
+export default function ruleSelectorRemover (ast, selectorNodeMap, selectors) {
+  selectors = new Set(selectors)
```

Comment:
This is ace - for large stylesheets this saves considerable bytes on the data needed to pass to the sandboxed function (to serialise) and hence makes Penthouse significantly faster! Will test in larger scale and come back with some numbers, but really excited about this! 💯

Reply:
I had noted before that just initialising the sandboxed function could take significant time for larger stylesheets but I never got down to exactly why and what could be done about this. So happy to see this happening now. :)

Reply:
It would be nice to know results of your benchmarks ;)

Reply:
So about 65% faster compared to previous version, as I shared in my tweet. It also seems to have resolved a small number of issues with generated criticalcss, presumably because of improved selector parsing. 👍